### PR TITLE
Only depend on beats when necessary

### PR DESCRIPTION
**Pull request by pugnascotia**
The unified build happens in stages, and for release builds the main
part of Elasticsearch will be built before the Beats distributions are
available. To work around this, infer in the Docker part of the build
whether to include beats as a dependency.  In the future, we should find
a more robust mechanism.

### CODE REVIEW

**Review by breskeby — CHANGES_REQUESTED** [review]
fine by me. In general I think we should avoid if statements in dependency blocks and instead differ at where this configuration is used. This make things more readable IMO

**Review by breskeby — APPROVED** [review]
lgtm

File: distribution/docker/build.gradle [hunk 1]
```diff
@@ -240,9 +240,17 @@ void addBuildDockerContextTask(Architecture architecture, DockerBase base) {
       }
 
       if (base == DockerBase.CLOUD) {
+        // If we're performing a release build, but `build.id` hasn't been set, we can
```
Comment (mark-vieira) [inline distribution/docker/build.gradle:243]: Question, so do we actually build the docker images multiple times?

Comment (pugnascotia) [reply distribution/docker/build.gradle:243]: Yes - we build the default image, the UBI image, the Iron Bank context, the Cloud Image and the Cloud ESS image. Apart from UBI and Iron Bank, we also build them on ARM.

**Pull request merged**